### PR TITLE
[AIRFLOW-3624] Add masterType parameter to MLEngineTrainingOperator

### PR DESCRIPTION
**Pull request by kkpoon**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-XXX
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.

### Description

Ref to document https://cloud.google.com/ml-engine/docs/tensorflow/machine-types

> When the scale_tier is set to CUSTOM, user should specify masterType
> The CUSTOM tier is not a set tier, but rather enables you to use your own cluster specification. When you use this tier, set values to configure your processing cluster according to these guidelines:
> 
> * You must set TrainingInput.masterType to specify the type of machine to use for your master node. This is the only required setting. See the machine types described below.

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.
  - All the public functions and the classes in the PR contain docstrings that explain what it does

### Code Quality

- [x] Passes `flake8`

### CODE REVIEW

File: airflow/contrib/operators/mlengine_operator.py [hunk 1]
```diff
@@ -473,6 +473,10 @@ class MLEngineTrainingOperator(BaseOperator):
     :param scale_tier: Resource tier for MLEngine training job. (templated)
     :type scale_tier: str
 
+    :param master_type: Cloud ML Engine machine name.
+        Must be set when scale_tier is CUSTOM. (templated)
```
Comment (jgao54) [inline airflow/contrib/operators/mlengine_operator.py:477]: to make master_type templated, it needs to be added to the class member `template_fields`

Comment (kkpoon) [reply airflow/contrib/operators/mlengine_operator.py:477]: Thanks @jgao54. It is updated

**Comment by jgao54** [comment]
Thanks! Could you update the commit message by prepending it with `[AIRFLOW-3624] `

**Comment by kkpoon** [comment]
@jgao54 the commit message is updated. Thanks

**Pull request merged**